posts/kotlin-like-java-but-better-2.md

```diff
@@ -0,0 +1,255 @@
+---
+authors:
+- Davis Mohar
+tags:
+- Java
+- Kotlin
+- Spring Boot
+date: 2015-05-17T12:21:50.000Z
+title: "Kotlin: Like Java, But Better (Part 2)"
+image: https://github.com/davismohar/blog-usa/blob/master/images/2021/05/kotlin-like-java-but-better-1.png
+---
+
+If you haven't already read part one of this blog series, then check out '[Kotlin: Like Java, But Better Part 1](https://blog.ippon.tech/kotlin-like-java-but-better-part-1/)' to learn about Kotlin's background and features!
+
+# Introduction
+One of Kotlin's greatest strengths is the full interoperability with Java. This enables developers to write Kotlin code in existing Java projects, which makes it much easier to get started with Kotlin. In this blog post, we will put this interoperability to the test by converting an existing Java Spring Boot microservice into Kotlin!
+
+If you're following along at home, you can clone the [source code from github](https://github.com/davismohar/kotlin-spring-boot-demo). The 'main' branch has the starting Java code, and the 'kotlin' branch contains the final app that has been converted to Kotlin. 
+
+It is highly recommend to use [IntelliJ IDEA](https://www.jetbrains.com/idea/) as your IDE when working with Kotlin files. 
+
+# Our Microservice
+First, lets take a tour of the Spring application we will be working with! This is a very basic app that has a few REST endpoints that allow us to manipulate a list of people. The endpoints are:
```

Comment:
This is a basic app that has a few REST endpoints that allow us to manipulate a list of people.